modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2665,7 +2665,16 @@ else if (selectExpression instanceof ScalarDSLQueryAlias) {
 							_getKey(), (String)objects[i]);
 					}
 					catch (Exception exception) {
-						throw new PortalException(exception);
+						if (_log.isDebugEnabled()) {
+							_log.debug(exception);
+						}
+
+						throw new IllegalArgumentException(
+							StringBundler.concat(
+								"Object entries are unavailable due to ",
+								"missing encryption key. Please upload an ",
+								"encryption key or remove the object's ",
+								"encryption field to recover these entries."));
```

Comment:
@p-albuquerque why did you choose this exception (`IllegalArgumentException`)? Why don't we use the `EncryptorException` instead?

Reply:
Actually, I don't try raise a **EncryptorException**. I choose this one, because in the described scenario, the exception is triggered when call `new SecretKeySpec(...`  [here](https://github.com/liferay/liferay-portal/blob/e8a34278ee2f44333f0dda69f407cffec7c4a1b4/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L1927-L1930)
this is the raised exception (external library):
![exception](https://github.com/liferay-objects/liferay-portal/assets/79872134/813a41f1-4489-49ff-b1f4-280b65261452)

Reply:
I just want to replace the message, so I though in override the message to the specific one, keeping the exception class.

Reply:
Can we change to:

```
catch (IllegalArgumentException illegalArgumentException) {
   throw new IllegalArgumentException(your message);
}
catch (Exception exception) {
   throw new PortalException(exception);
}
```
?